gapic/ads-templates/%namespace/%name/%version/%sub/services/%service/client.py.j2

```diff
@@ -148,6 +148,18 @@ class {{ service.client_name }}(metaclass={{ service.client_name }}Meta):
         """
         return self._transport
 
+    def __enter__(self):
+        self.transport.__enter__()
+        return self
+
+    def __exit__(self, type, value, traceback):
+        """Releases underlying transport's resources.
+
+        WARNING: ONLY use as a context manager if the transport is NOT shared
```

Comment:
Will this appear anywhere a developer can see it in the documentation? An explicit `close()` method could be useful, even if it's sole purpose is to document that you shouldn't close something where you don't want the underlying transport to be closed.

Reply:
I don't believe we currently generate docs for this, so this would only show up in `doctsring`. I don't like `close` since it assumes that the object needs to be `opened`, which doesn't make sense for a client. Having both `__exit__` and `close` would be redundant too.

Reply:
@tswast Is the concern that `__exit__` won't show up in the generated refdocs?

Reply:
That is my primary concern, yes.

Reply:
I disagree that one needs `open()` for there to be a `close()`, but I would expect most people to use `__exit__` if available. There is utility in having an explicit `close()` method, but my main concern is the discoverability in the docs.

Reply:
`close()` would allow some flexibility that just a context manager doesn't allow. For example, see the PR some customers have linked to this issue: https://github.com/coiled/dask-bigquery/pull/4#discussion_r710418038

Reply:
Let's keep `close()` from client for now. If this problem arises, we can follow-up in separate CL.

Reply:
@atulep We still need to make sure this warning appears somewhere visible in the docs. When you generate the docs for one of these clients, do you see `__exit__` in the documentation?

Reply:
It shows up in generated reference. I also added an appropriate infobox around warning to make it stand out.